File: mkdocs/snippets/devbook/accounts/create-from-private-key.py
import os

from symbolchain.CryptoTypes import PrivateKey
from symbolchain.facade.SymbolFacade import SymbolFacade

# Initialize the facade for the testnet network [>step-1]
facade = SymbolFacade('testnet')
# [<step-1]
# Use an existing private key if provided, [>step-2]
# Otherwise generate a random one.
private_key_string = os.getenv('PRIVATE_KEY')
if private_key_string:
	print("Loading account from environment variable...")
	private_key = PrivateKey(private_key_string)
else:
	print("Generating random account...")
	private_key = PrivateKey.random()
# [<step-2]
# Create a key pair from the private key [>step-3]
key_pair = facade.KeyPair(private_key)

# Derive the public key from the private key
public_key = key_pair.public_key

# Derive the address from the public key
address = facade.network.public_key_to_address(public_key)

# Output the account details
print(f'Address: {address}')
print(f'Public key: {public_key}')
print(f'Private key: {private_key}') # [<step-3]

File: mkdocs/snippets/devbook/accounts/CreateFromPrivateKey.java
//JAVA 21+
//DEPS org.symbol:symbol-sdk:3.3.1

import org.symbol.sdk.CryptoTypes;
import org.symbol.sdk.facade.SymbolFacade;
import org.symbol.sdk.symbol.Address;
import org.symbol.sdk.symbol.KeyPair;

public final class CreateFromPrivateKey {
	// Initialize the facade for the testnet network [>step-1]
	private final SymbolFacade facade = new SymbolFacade("testnet");
	// [<step-1]

	public static void main(final String[] args) {
		new CreateFromPrivateKey().run();
	}

	private void run() {
		// Use an existing private key if provided, [>step-2]
		// Otherwise generate a random one.
		final String privateKeyString = System.getenv("PRIVATE_KEY");
		final CryptoTypes.PrivateKey privateKey;
		if (null != privateKeyString) {
			System.out.println(
				"Loading account from environment variable...");
			privateKey = new CryptoTypes.PrivateKey(privateKeyString);
		} else {
			System.out.println("Generating random account...");
			privateKey = CryptoTypes.PrivateKey.random();
		} // [<step-2]

		// Create a key pair from the private key [>step-3]
		final KeyPair keyPair = new KeyPair(privateKey);

		// Derive the public key from the private key
		final CryptoTypes.PublicKey publicKey = keyPair.getPublicKey();

		// Derive the address from the public key
		final Address address = facade.network.publicKeyToAddress(
			publicKey);

		// Output the account details
		System.out.printf("Address: %s%n", address);
		System.out.printf("Public key: %s%n", publicKey);
		System.out.printf("Private key: %s%n", privateKey); // [<step-3]
	}
}
